The band-image-analysis branch was updated (#6246)
galaxyproject/training-material: topics/imaging/tutorials/electrophoresis-gel-bands-image-analysis/tutorial.md

> <agenda-title></agenda-title>
>
> The steps covered in this tutorial are:
>
> 1. Uploading and preparing the electrophoresis gel image  
> 2. Selecting and exporting regions of interest (ROIs) using QuPath  
> 3. Converting ROI coordinates to a label map  
> 4. Quantifying band intensity  
> 5. Inspecting results
>
{: .agenda}

# Data preparation

The example dataset can be downloaded from here [Electrophoresis gel dataset](https://github.com/galaxyproject/training-material/tree/main/topics/imaging/images/electrophoresis-gel-bands-image-analysis/Image.jpg).

![Electrophoresis Gel](../../images/electrophoresis-gel-bands-image-analysis/Image.jpg)

This image was published by [Kenji Ohgane et al. (2019) on Protocols.io](https://www.protocols.io/view/quantification-of-gel-bands-by-an-image-j-macro-ba-bp2l6n4bkgqe/v1).

> <hands-on-title>Download and upload the image</hands-on-title>
>
> 1. Create a new history in Galaxy.  

> > 2. Download the test image using the link above.  

Author: @shiltemann
Thank you for reviewing my PR and for your feedback. I appreciate your support and guidance.

> > 2. Download the test image using the link above.  

Author: @shiltemann 
Thank you for your suggestion! I have created a Zenodo link for all the data related to this tutorial and have given you access. Would you please check it.

> 3. Click the **{% icon galaxy-upload %} Upload** to add the image file into your galaxy history.  
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>    {% snippet faqs/galaxy/datasets_import_from_data_library.md %}
>
>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="datatypes" %}
>
> 4. Rename {% icon galaxy-pencil %} the dataset to `Electrophoresis-Gel.jpg`.
>
{: .hands_on}

> Before we dive into the quantification steps let's convert our jpg image file into the image format TIFF. This format is preferred for analysis and quantification as it preserves the image data and metadata. 

Reviewer: Please use consistent capitalization for file formats, like `jpg` _and_ `tiff` or `JPG` _and_ `TIFF`:

```suggestion
Before we dive into the quantification steps let's convert our JPG image file into the TIFF image format. This format is preferred for analysis and quantification as it preserves the image data and metadata. 
```

Reading IFDs
Populating metadata
Checking comment style
Populating OME metadata
Initialization took 0.569s

Reading core metadata
filename = /data/dnb11/galaxy_db/files/0/6/6/dataset_06685fae-ac3e-48b9-a460-d0ad39ea07a6.dat
Series count = 1
Series #0 :
	Image count = 1
	RGB = true (3) 
	Interleaved = false
	Indexed = false (false color)
	Width = 1278
	Height = 250
	SizeZ = 1
	SizeT = 1
	SizeC = 3 (effectively 1)
	Thumbnail size = 128 x 25
	Endianness = intel (little)
	Dimension order = XYCZT (certain)
	Pixel type = uint8
	Valid bits per pixel = 8
	Metadata complete = true
	Thumbnail series = false
	-----
	Plane #0 <=> Z 0, C 0, T 0

Reading global metadata
BitsPerSample: 8
Compression: JPEG
Document Name: temp.tiff
ImageLength: 250
ImageWidth: 1278
MetaDataPhotometricInterpretation: RGB
MetaMorph: no
NumberOfChannels: 3
PageNumber: 0 1
PhotometricInterpretation: YCbCr
PlanarConfiguration: Chunky
ReferenceBlackWhite: 0
SampleFormat: unsigned integer
SamplesPerPixel: 3
Software: GraphicsMagick 1.3.45 2024-08-27 Q8 http://www.GraphicsMagick.org/

Reading metadata
  </pre>
</div>

## Split image into separate channels

> TIFF images can store multiple channels—such as different colors or fluorescence signals—in a single file. In this step, we will separate these channels by splitting the image along the channel axis (C-axis). This results in a set of single-channel images, which are required for the next processing steps.

Reviewer: Somewhat nitpicky, but strictly speaking, the result will be a list:

```suggestion
TIFF images can store multiple channels—such as different colors or fluorescence signals—in a single file. In this step, we will separate these channels by splitting the image along the channel axis (C-axis). This results in a list of single-channel images, which are required for the next processing steps.
```

Canonically, sets are not associated with ordering, but here the elements have a strict order, right?

Author: @kostrykin 
Yes, of course! Thank you for your suggestion.

> <hands-on-title>Split image into single channels</hands-on-title>
> 1. {% tool [Split image along axes](toolshed.g2.bx.psu.edu/repos/imgteam/split_image/ip_split_image/2.2.3+galaxy1) %} with the following parameters:
>    - {% icon param-file %} *"Image to split"*: Select your uploaded `Electrophoresis-Gel.tiff` file
>    - *"Axis to split along"*: Select `C-axis (split the channels of an image or image sequence)` from the dropdown menu
>
{: .hands_on}

> > <comment-title>Use a split image for downstream compatibility and annotation</comment-title>

Reviewer: ```suggestion
> <comment-title>Use a single-channel image for downstream compatibility and annotation</comment-title>
```

> While QuPath can handle multi-channel images, our other tools cannot. Therefore, after splitting the image, we will use the {% tool [Extract Dataset](__EXTRACT_DATASET__) %} tool to select one of the single-channel images. This image should represent the intensity channel we want to measure, and will be used for QuPath regions of interest (ROI) annotation.
>
{: .comment}

> <hands-on-title>Extract the TIFF file from collection</hands-on-title>
>
> 1. {% tool [Extract Dataset](__EXTRACT_DATASET__) %} with the following parameters:
>    - {% icon param-collection %} *"Input Collection"*: Select the collection output from the `Split Image` tool
>    - *"Which part of the Collection?"*: Choose the TIFF file that represent the channel of your interest. In our dataset example, we extract 1.tiff
>
{: .hands_on}

> 

Reviewer: ```suggestion
{: .hands_on }

```

> 

Reviewer 2: > 

Thanks a lot @shiltemann!! For this comment, please take my suggestion instead of this one because I made the mistake to keep this line > 2. {% icon galaxy-pencil %} **Rename** the extracted file to `Bands.geojson`. It is not needed in the hands-on

> 

Reviewer 3: @dianichj cool, I will mark this one as resolved :+1:
